closes #2: address circleci feedback

eloypnd opened the pull request:
**SEMVER Update Type:**
- [ ] Major
- [ ] Minor
- [x] Patch

## Description:

- remove executor
- change token parameter type
- add example for workflow with test changes

## Motivation:

Address feedback from CircleCI Team

 **Closes Issues:**
-  Closes #2

## Checklist:

- [x] All new jobs, commands, executors, parameters have descriptions.
- [x] Usage Example version numbers have been updated.

mikkottis commented on src/examples/schedule-run.yml:
```diff
@@ -0,0 +1,16 @@
+    schedule-run:
+      jobs:
+        - testlio/schedule-run:
+            requires:
+              - fetch-functional-tests
```
Leftover?

huzaifaAhmedTestlio commented on src/examples/schedule-run-for-a-project.yml:
```diff
@@ -1,14 +1,35 @@
 usage:
   version: 2.1
   orbs:
     testlio: testlio/testlio@1.0.0
```
In what case we'll be bumping the minor version?

KyleTryon commented on src/examples/schedule-run-fetching-tests.yml:
```diff
@@ -0,0 +1,35 @@
+            test-config: "./functional-tests-folder/test-config.json"
+            project-config: "./functional-tests-folder/project-config.json"
+
+  fetch-functional-tests:
```
This belongs under a `jobs` key.

KyleTryon replied:
This job also has no executor,

example:
```
docker:
  - image: cimg/node:lts
```

eloypnd replied:
Good catch @KyleTryon, thanks!

I have fixed it. PTAL

KyleTryon reviewed (changes requested):
Looks good. I just noticed two small issues with one of the usage examples, other than that I would say it is good to go!

KyleTryon commented:
Hello @eloypnd, sorry for the delay in getting back to you. 

The changes look much better, though I still have one question from the original review:

> * Job schedule-run
Attaching the workspace appears to be optional, and off by default. Without a workspace, will a checkout step be needed?
> * Usage example
In your usage example, the jobs is called in a workflow and I see attach workspace is set to true, but there are no other jobs in the workflow. Should there be a sample of a job that would run before schedule-run?

eloypnd commented:
Hi @KyleTryon, thanks for your questions.

Usually, when running functional tests there are 2 common cases:

1. The functional tests code is in the same codebase as the program one. That's the case for this example: https://github.com/Testlio/testlio-circleci-orb/pull/3/files#diff-4a53abf8681f77c5f131a2b21f906d6027939bff602598387e5f07b15c7ced10. But I realised now that in this example we are missing the `checkout` step.

2. The functional tests code is in a different codebase and it needs to be fetched in order to run it, like in this other example: https://github.com/Testlio/testlio-circleci-orb/pull/3/files#diff-5e6a04928448c4b1435426bcd3746ffb615db1c9579622b0195eed12a8fbe635. In this case we depend on a previous job that fetches the tests code and that's when we need to attach the workspace.

I will be pushing a new commit to add the `checkout` step into example 1. Let me know if that clarifies your questions or you need some more details.

KyleTryon commented:
Hello @eloypnd 

It looks like in the most recent change the _step_ "checkout" was added to the workflow as if it were a job. The checkout step would need to be a part of the job itself.

EricRibeiro commented:
Thanks for addressing the issues Kyle pointed out @eloypnd.

The `testlio/schedule-run` job [prints](https://app.circleci.com/pipelines/github/Testlio/testlio-circleci-orb/50/workflows/791cb969-de63-420e-bf79-4b882e05bd0b/jobs/385?invite=true#step-103-20):

```
Something went wrong while scheduling the automated run SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at module.exports (/home/circleci/.nvm/versions/node/v16.15.1/lib/node_modules/@testlio/cli/bin/create-run.js:349:18)
    at Object.<anonymous> (/home/circleci/.nvm/versions/node/v16.15.1/lib/node_modules/@testlio/cli/bin/index.js:17:26)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47
```

I'm assuming this is because you are using dummy files. Would you be able to run this test with valid files?

eloypnd commented:
> Thanks for addressing the issues Kyle pointed out @eloypnd.
> 
> The `testlio/schedule-run` job [prints](https://app.circleci.com/pipelines/github/Testlio/testlio-circleci-orb/50/workflows/791cb969-de63-420e-bf79-4b882e05bd0b/jobs/385?invite=true#step-103-20):
> 
> ```
> Something went wrong while scheduling the automated run SyntaxError: Unexpected end of JSON input
>     at JSON.parse (<anonymous>)
>     at module.exports (/home/circleci/.nvm/versions/node/v16.15.1/lib/node_modules/@testlio/cli/bin/create-run.js:349:18)
>     at Object.<anonymous> (/home/circleci/.nvm/versions/node/v16.15.1/lib/node_modules/@testlio/cli/bin/index.js:17:26)
>     at Module._compile (node:internal/modules/cjs/loader:1105:14)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
>     at Module.load (node:internal/modules/cjs/loader:981:32)
>     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
>     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
>     at node:internal/main/run_main_module:17:47
> ```
> 
> I'm assuming this is because you are using dummy files. Would you be able to run this test with valid files?

Thanks @EricRibeiro for your feedback.

We have implemented a `dryRun` option for our CLI that we can use to test the Orb in this case. Now the `testlio/schedule-run` job [prints](https://app.circleci.com/pipelines/github/Testlio/testlio-circleci-orb/56/workflows/588b0069-c3b8-4bfd-a5e0-167c6f40c48a/jobs/434).

```
Dry run mode. Run creation omitted.
```

And we have added two "sample config files" that are validated in the `dryRun` mode:
- [sample-config/project-config.json](https://github.com/Testlio/testlio-circleci-orb/pull/3/files#diff-54ecd52885dac0919a62448a3e7f3abce0febb4f21c748e559c3baa8989c1b17)
- [sample-config/test-config.json](https://github.com/Testlio/testlio-circleci-orb/pull/3/files#diff-a16902db5d9c8d12b08260ce0bf2b66a479120e6214ef3ed4d6b40fee533a195)

Please take a look and let me know if you have any extra comments or we have addressed all the issues.

aldomozhirov commented on src/scripts/create-run.sh:
```diff
@@ -8,4 +8,9 @@ set -o xtrace
+if [ "$DRY_RUN" -eq 1 ] ; then
+    testlio create-run --dryRun --projectConfig ${PROJECT_CONFIG_PATH} --testConfig ${TEST_CONFIG_PATH}
+else
+    testlio create-run --projectConfig ${PROJECT_CONFIG_PATH} --testConfig ${TEST_CONFIG_PATH}
```
Shouldn't we also put --pollResults flag to wait until results will appear?

eloypnd replied:
We can handle adding this new option as a separate feature request, as far as is not directly related with the feedback from #2 that this PR is addressing.

EricRibeiro commented on src/scripts/create-run.sh:
```diff
@@ -8,4 +8,9 @@ set -o xtrace
-testlio create-run --projectConfig ${PROJECT_CONFIG_PATH} --testConfig ${TEST_CONFIG_PATH}
+
+if [ "$DRY_RUN" -eq 1 ] ; then
+    testlio create-run --dryRun --projectConfig ${PROJECT_CONFIG_PATH} --testConfig ${TEST_CONFIG_PATH}
```
`${PROJECT_CONFIG_PATH}` and `${TEST_CONFIG_PATH}`  should be double-quoted to avoid unexpected behaviour.

eloypnd replied:
Good point. I've made the changes: https://github.com/Testlio/testlio-circleci-orb/pull/3/commits/41c59892990ba2780955098ec0b05c9fab31642a

EricRibeiro commented on src/scripts/create-run.sh:
```diff
@@ -8,4 +8,9 @@ set -o xtrace
+if [ "$DRY_RUN" -eq 1 ] ; then
+    testlio create-run --dryRun --projectConfig ${PROJECT_CONFIG_PATH} --testConfig ${TEST_CONFIG_PATH}
+else
+    testlio create-run --projectConfig ${PROJECT_CONFIG_PATH} --testConfig ${TEST_CONFIG_PATH}
```
Same as above.
